pkg/loadbalancer/smooth_weighted_round_robin_test.go

```diff
@@ -0,0 +1,310 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package loadbalancer
```

Comment:
I've found it's usually best practice to keep unit tests "whitebox" and use the public interface and avoid exposing internals.

```suggestion
package loadbalancer_test
```

Reply:
Yea, but it contradicts my need to validate the internal map and list which I used the `All()` function...
How would you suggest I do so? Should I remove the tests for the internal structures? or write separate internal tests for them?

Reply:
You shouldn't need to explicitly access the internals - test what's observed via the public interface. Basically you `Add` item(s) and verify they're returned correctly via `Next` based on the weights. You have tests that do that. `RemoveAll` and `Skip` can also be verified via `Next`, as you've done.